src/libs/antares/study/binding_constraint/BindingConstraintLoader.cpp

```diff
@@ -5,9 +5,36 @@
 #include "BindingConstraintLoader.h"
 #include <memory>
 #include <vector>
+#include <sstream>
 #include "BindingConstraint.h"
 #include "yuni/core/string/string.h"
 #include "antares/study/version.h"
+#include "antares/constants.h"
+
+bool hoursToVector(std::vector<bool>& output,
+                   const std::string& input)
+{
+    output.assign(Antares::Constants::nbHoursInAWeek, false);
+
+    std::istringstream ss(input);
+    while (ss) {
+        std::string token;
+        if (std::getline(ss, token, ',')) { // Split by comma
+            try {
+                unsigned int index = std::stoi(token) - 1;
```

Comment:
test the return value of stoi in case of nonconvertible str and use the more robust  function [strtol](https://cplusplus.com/reference/cstdlib/strtol/)

Reply:
Why is it more robust ? It's a C function

Reply:
i don't know, why i read atoi, 👎🏽  for me

Reply:
probably i was thinking about this cases:

`
#include <iostream>
#include <string>
#include <cstdlib>
#include <ios>
using namespace std;
void test_str(std::string s){
    errno = 0;
    cout<<"convert to int: "<<s<<"\n";
    
    
    char* ptr_end{};
    const char*c_s =s.c_str(); 
    cout<<"strtol\n";
    auto  j = std::strtol(c_s, &ptr_end, 10);
    cout<<"j: "<<j<<"\n";
    cout<<"ptr_end: "<<ptr_end<<"\n";
    cout << "ptr_end == c_s: "<<std::boolalpha<<  (c_s == ptr_end)<<"\n";
    
    
    cout<<"stoi\n";
    auto  i = std::stoi(s);
    cout<<"i: "<<i<<"\n";
    cout<<"**************************\n";
    
    
}
int main()
{
    test_str("123abc");
    test_str("123");
    test_str("");

    return 0;
}`

Reply:
but irrelevant since it make the code harder to read